Author: Diane Trout:
Description: There is a problem where the returned dtype is still a 64
bit integer when returned from sql even on 32 bit platforms. This
leads the assert_eq to fail because of the differing dtype. Upstream
first encountered this on win32, but Debian still also builds 32-bit
linux architectures. I change the check_dtype test to ignore any
32-bit architecture

--- a/dask/dataframe/io/tests/test_sql.py
+++ b/dask/dataframe/io/tests/test_sql.py
@@ -438,7 +438,7 @@
     ).select_from(sql.table("test"))
     out = read_sql_query(s1, db, npartitions=2, index_col="number", meta=meta)
     # Don't check dtype for windows https://github.com/dask/dask/issues/8620
-    assert_eq(out, df[["name", "age"]], check_dtype=sys.platform != "win32")
+    assert_eq(out, df[["name", "age"]], check_dtype=sys.int_info.sizeof_digit != 2)
 
 
 def test_no_character_index_without_divisions(db):
